File: frontend/types/shop.ts
// ==================== ENUMS ====================

export enum RegistrationStatus {
  Pending = "Pending",
  Approved = "Approved",
  Rejected = "Rejected",
}

export enum ShopStatus {
  Inactive = "Inactive",
  Active = "Active",
  Suspended = "Suspended",
}

// ==================== DTOs ====================

/** Khớp với BusinessRegistrationDto (BE) */
export interface BusinessRegistrationDto {
  id: number;
  userId: string;
  companyName: string;
  taxCode: string;
  ownerName: string;
  email: string;
  phone: string;
  address: string;
  status: string; // "Pending" | "Approved" | "Rejected"
  rejectReason: string | null;
  createdAt: string;
  reviewedAt: string | null;
  reviewedBy: string | null;
}

/** Khớp với ShopDto (BE) */
export interface ShopDto {
  id: number;
  ownerUserId: string;
  businessRegistrationId: number;
  name: string;
  slug: string;
  description: string | null;
  logoUrl: string | null;
  coverUrl: string | null;
  status: string; // "Inactive" | "Active" | "Suspended"
  createdAt: string;
  updatedAt: string | null;
}

// ==================== REQUEST PAYLOADS ====================

/** Khớp với CreateBusinessRegistrationCommand (BE) */
export interface CreateBusinessRegistrationRequest {
  companyName: string;
  taxCode: string;
  ownerName: string;
  email: string;
  phone: string;
  address: string;
}

/** Khớp với ReviewBusinessRegistrationDto (BE) */
export interface ReviewRegistrationRequest {
  rejectReason?: string;
}

/** Khớp với UpdateShopCommand (BE) */
export interface UpdateShopRequest {
  id: number;
  name: string;
  slug: string;
  description?: string;
  logoUrl?: string;
  coverUrl?: string;
}

File: backend/ShopxBase.Domain/Entities/Shop.cs
using ShopxBase.Domain.Enums;

namespace ShopxBase.Domain.Entities;

public class Shop : BaseEntity
{
    public string OwnerUserId { get; set; }
    public int BusinessRegistrationId { get; set; }
    public string Name { get; set; }
    public string Slug { get; set; }
    public string? Description { get; set; }
    public string? LogoUrl { get; set; }
    public string? CoverUrl { get; set; }
    public ShopStatus Status { get; set; } = ShopStatus.Inactive;

    public virtual BusinessRegistration BusinessRegistration { get; set; }
    public virtual ICollection<Product> Products { get; set; } = new List<Product>();
    public virtual ICollection<ShopMember> Members { get; set; } = new List<ShopMember>();
}
